### ISSUE TITLE
Tasks are still in IN_PROGRESS/SCHEDULED state even though workflow/sub-workflow are in FAILED/TERMINATED state

### ISSUE THREAD

**Issue by bharadwajrembar** [Netflix/conductor#2117]
Hi,

We are running Conductor v2.26.0 with Elasticache Redis cluster as our metadata store and Elasticsearch 6.x as our indexing store.

We also leverage dynamic forks and sub-workflows heavily in our project. We notice that when sub-workflows fail, they don't actually put their tasks in a CANCELLED or a FAILED state. Rather they are still SCHEDULED/IN_PROGRESS.

Because of this, our workers tend to poll tasks from workflows that are already in an end state unnecessarily.

When they are polled and updated by our workers, we see this in the Conductor server logs:

```
[2021-03-08T05:43:01.840Z] INFO qtp652481263-19664 WorkflowExecutor:799 Workflow: <REDACTED SUB-WORKFLOW NAME>.1/3dce8785-d71f-45f2-9f35-7ca4bef39082.TERMINATED has already finished execution. Task update for: 8c154c7d-3be2-48c3-bfa0-227dd9370946 ignored and removed from Queue: <REDACTED TASK NAME>.
```

Is there any solution/fix for this?

**Comment by aravindanr** [comment]
hi @bharadwajrembar, thanks for reporting this. We have also identified some glitches with subworkflows inside a fork join. There is a fix on the way for 3.0 which will be back-porting to 2.30. I am marking this as a bug and will update this thread when the fix is released.

**Comment by bharadwajrembar** [comment]
Sure @aravindanr thanks for this

**Comment by indiv0** [comment]
@aravindanr Hi, is there any progress on the fix? Either in 3.0 or as a back port? It appears that we might be encountering this bug as well.

**Comment by apanicker-nflx** [comment]
This should be fixed as part of #2125 in v3.x 
Given the expanse of the changes, it was decided to not backport to 2.31 at the time. However, if you have the bandwidth we would welcome a PR with the changes.
